subprojects/docs/src/docs/userguide/javaLibraryPlugin.adoc

```diff
@@ -55,6 +55,12 @@ Dependencies appearing in the `api` configurations will be transitively exposed
 The `compile` configuration still exists but should not be used as it will not offer the guarantees that the `api` and `implementation` configurations provide.
 ====
 
+If your build consumes a published module with POM metadata, the Java and Java Library plugins both honor api and implementation separation through the scopes used in the pom. Meaning that the compile classpath only includes `compile` scoped dependencies, while the runtime classpath adds the `runtime` scoped dependencies as well. While this often does not have an effect for modules published with Maven, it can have a big impact for modules developed with the Java Library plugin and published with Gradle.
```

Comment:
I might be missing context, but the end of this paragraph seems serious ... but not very specific. Would it make sense to add an example of the impact it can cause?

Reply:
I don't quite understand the last sentence either. It would be helpful tell the user when exactly a problematic situation  can occur.

Reply:
There is more context above in this section. But yes. More explanation here wouldn't hurt. I expanded the sentence into an explaining paragraph.

All in all, I think whole section on Java Libraries should be checked and improved. But I won't do that as part of this PR. (I think it's on you are planing to work on that Ben, right?)